lib/rubocop/cop/style/braces_around_hash_parameters.rb

```diff
@@ -92,31 +92,65 @@ def autocorrect(send_node)
 
           lambda do |corrector|
             if hash_node.braces?
-              remove_braces_with_whitespace(corrector, hash_node)
+              remove_braces_with_whitespace(corrector,
+                                            hash_node,
+                                            extra_space(hash_node))
             else
               add_braces(corrector, hash_node)
             end
           end
         end
 
-        def remove_braces_with_whitespace(corrector, node)
-          right_brace_and_space = right_brace_and_space(node.loc.end)
+        def extra_space(hash_node)
+          {
+            newlines: extra_left?(hash_node) && extra_right?(hash_node),
+            left: extra_left?(hash_node),
+            right: extra_right?(hash_node)
+          }
+        end
+
+        def extra_left?(hash_node)
+          @extra_left ||= begin
+            top_line = hash_node.source_range.source_line
```

Comment:
I think in most of the code we've used `first_line` and `last_line` in similar cases.

Reply:
`first_line` reports the line number while `source_line` delivers the actual content. I could switch it over to mimic how `extra_right?` uses `last_line`. But then I'd have to [add an operation on the `processed_source`](https://github.com/bbatsov/rubocop/pull/5066/files#diff-f688fd4df1d0129b33ff64cea9c08d1cR122).